File: app/src/main/java/com/yusril/tugaswidget/WidgetAnalogClock.java
package com.yusril.tugaswidget;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.AnalogClock;
import android.widget.Button;

@SuppressWarnings("ALL")
public class WidgetAnalogClock extends AppCompatActivity {
    AnalogClock analogClock;
    Button button;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_widget_analog_clock);
        analogClock = (AnalogClock) findViewById(R.id.analogClock);
        button = (Button) findViewById(R.id.button);
        analogClock.setVisibility(View.INVISIBLE);
    }

    public void onButtonClicked(View view) {
        button.setText("Analog Clock");
        analogClock.setVisibility(View.VISIBLE);
    }
}

File: app/src/main/java/com/yusril/tugaswidget/WidgetAnalogClock.kt
@file:Suppress("DEPRECATION")

package com.yusril.tugaswidget

import android.annotation.SuppressLint
import androidx.appcompat.app.AppCompatActivity

import android.os.Bundle
import android.view.View
import android.widget.AnalogClock
import android.widget.Button

class WidgetAnalogClock : AppCompatActivity() {
    private lateinit var analogClock: AnalogClock
    internal lateinit var button: Button
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_widget_analog_clock)
        analogClock = findViewById<View>(R.id.analogClock) as AnalogClock
        button = findViewById<View>(R.id.button) as Button
        analogClock.visibility = View.INVISIBLE
        if (supportActionBar != null) {
            supportActionBar!!.setBackgroundDrawable(resources.getDrawable(R.drawable.gradasi))
        }
    }

    @SuppressLint("SetTextI18n")
    fun onButtonClicked() {
        button.text = "Analog Clock"
        analogClock.visibility = View.VISIBLE
    }
}
